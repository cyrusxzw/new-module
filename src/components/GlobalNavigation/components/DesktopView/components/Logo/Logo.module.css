@import 'styles/index';

.base {
  position: absolute;
  z-index: 20;
  top: calc(80px + 74px + 25px);
  left: 40px;
  padding: 5px;
  pointer-events: auto;
  transform: translateX(-5px);

  &.active {
    top: calc(80px + 74px);
  }

  &.dark:focus:not(:hover) {
    outline: 1px dashed var(--color-dark-decoration);
    outline-offset: 2px;
  }

  &.light:focus:not(:hover) {
    outline: 1px dashed var(--color-light-decoration);
    outline-offset: 2px;
  }
}

.icon {
  &.dark {
    fill: var(--color-secondary);
  }

  &.light {
    fill: var(--color-primary);
  }

  .active & {
    fill: var(--color-secondary);
  }
}
