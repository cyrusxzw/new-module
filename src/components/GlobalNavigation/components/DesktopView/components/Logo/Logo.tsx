import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import { ScreenReaderOnly } from '~/components/ScreenReaderOnly';
import {
  useGlobalNavigationContext,
  useGlobalNavigationStateContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';
import type { LogoType } from './Logo.types';
import styles from './Logo.module.css';

const Logo: LogoType = () => {
  const { isOpen } = useGlobalNavigationStateContext();
  const {
    actions: {
      logo: { dataTestRef, title, url, label },
    },
    desktopViewLogoTheme = 'dark',
  } = useGlobalNavigationContext();

  const currentTheme = useThemeContext(undefined, 'dark');

  const classSet = cx(styles.base, styles[currentTheme], {
    [styles.active]: isOpen,
  });

  const iconClassSet = cx(styles.icon, styles[desktopViewLogoTheme]);

  return (
    <Hyperlink
      className={classSet}
      dataTestRef={dataTestRef ?? 'NAV_LOGO'}
      title={title}
      url={url}
    >
      <Icon
        className={iconClassSet}
        height={40}
        name="aesop"
        tabIndex={-1}
        width={120}
      />

      <ScreenReaderOnly>{label}</ScreenReaderOnly>
    </Hyperlink>
  );
};

export { Logo };
