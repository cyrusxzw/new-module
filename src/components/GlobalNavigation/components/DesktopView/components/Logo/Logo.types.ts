import type { Themes, ComponentWithChildren } from '~/types';

type LogoProps = {
  className?: string;
  theme?: Themes;
};

type LogoType = ComponentWithChildren<LogoProps>;

export type { LogoType };
