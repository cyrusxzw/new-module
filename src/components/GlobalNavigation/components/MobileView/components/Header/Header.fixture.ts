const HeaderFixture = {
  isActive: true,
  onCloseButtonClick: () => undefined,
};

export { HeaderFixture };
