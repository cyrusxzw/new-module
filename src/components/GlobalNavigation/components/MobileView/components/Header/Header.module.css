@import 'styles/index';

.base {
  background-color: transparent;
}

.list {
  display: flex;
  height: 90px;
  align-items: center;
  justify-content: space-between;
  padding-top: 20px;
  padding-right: 20px;
  padding-bottom: 20px;
  padding-left: 20px;
  margin-top: 0;
  margin-bottom: 0;
  list-style: none;
}

.item {
  &:not(:first-child) {
    margin-left: 12px;
  }

  &:not(:last-child) {
    margin-right: 12px;
  }
}

.action {
  transition: color 200ms var(--easing-ease-out-cubic),
    outline-offset 200ms var(--easing-ease-out-cubic);

  .dark &:focus-visible,
  .darkClosed &:focus-visible {
    outline: 1px dashed var(--color-dark-decoration);
    outline-offset: 2px;
  }

  .light &:focus-visible,
  .lightClosed &:focus-visible {
    outline: 1px dashed var(--color-light-decoration);
    outline-offset: 2px;
  }
}

.itemLogo {
  flex-grow: 1;
}

.actionSearch {
  display: inline-flex;
  padding: 5px;
  font-family: var(--font-family-sans-serif-regular);
  line-height: 1;
  transform: translateY(-5px);
}

.searchIcon {
  .dark &,
  .darkClosed & {
    path {
      fill: var(--color-secondary);
    }
  }

  .light &,
  .lightClosed & {
    path {
      fill: var(--color-primary);
    }
  }
}

.actionCart {
  display: inline-flex;
  padding: 5px;
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(14px);
  line-height: 1;
  transform: translateY(-5px);

  .dark &,
  .darkClosed & {
    color: var(--color-secondary);
  }

  .light &,
  .lightClosed & {
    color: var(--color-primary);
  }
}

.ornamentalHover::after {
  .darkClosed & {
    background-color: var(--color-secondary);
  }

  .lightClosed & {
    background-color: var(--color-primary);
  }
}

.itemMenu:last-child {
  margin-left: 7px;
}

.actionMenu {
  width: 16px;
  height: 16px;
  box-sizing: content-box;
  padding: 5px;
  transform: translate(5px, -7px);

  &::before {
    top: 9px;
  }

  &::after {
    top: 16px;
  }

  &::before,
  &::after {
    position: absolute;
    left: 5px;
    width: 16px;
    height: 2px;
    content: '';
    transition: all 200ms var(--easing-ease-out-cubic);
  }

  &.openState::before {
    transform: rotate(45deg);
  }

  &.openState::after {
    transform: rotate(-45deg);
  }

  &.openState::after,
  &.openState::before {
    top: 12px;
  }

  .dark &::before,
  .dark &::after,
  .darkClosed &::before,
  .darkClosed &::after {
    background-color: var(--color-secondary);
  }

  .light &::before,
  .light &::after,
  .lightClosed &::before,
  .lightClosed &::after {
    background-color: var(--color-primary);
  }
}

.search {
  z-index: calc(var(--z-index-fixed-menu) + 1);
  top: 90px;
  height: 100%;

  .dark & {
    border-top: 1px solid var(--color-grey-60);
    background-color: var(--color-primary);
  }

  .light & {
    border-top: 1px solid var(--color-white);
    background-color: var(--color-secondary);
  }
}
