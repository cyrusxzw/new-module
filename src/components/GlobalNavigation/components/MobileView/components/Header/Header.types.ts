import type { ComponentWithoutChildren } from '~/types';
import type { MenuItemTrackingWithMobileAction } from '~/components/GlobalNavigation/GlobalNavigation.types';

type HeaderProps = {
  onClose: () => void;
};

type HeaderType = ComponentWithoutChildren<HeaderProps>;

type MenuItemTrackingWithMobileActionType = MenuItemTrackingWithMobileAction;

export type { HeaderType, HeaderProps, MenuItemTrackingWithMobileActionType };
