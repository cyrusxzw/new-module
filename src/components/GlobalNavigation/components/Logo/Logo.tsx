import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import {
  useGlobalNavigationContext,
  useGlobalNavigationStateContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import { ScreenReaderOnly } from '~/components/ScreenReaderOnly';
import { getDimentionsByView } from './Logo.utils';
import type { LogoType } from './Logo.types';
import styles from './Logo.module.css';

const Logo: LogoType = ({ closedTheme, isVisible = true }) => {
  const { isOpen, activeView } = useGlobalNavigationStateContext();
  const {
    actions: {
      logo: { dataTestRef, title, url, label },
    },
  } = useGlobalNavigationContext();

  const currentTheme = useThemeContext(undefined, 'dark');
  const currentClosedTheme = closedTheme || currentTheme;

  const { width, height } = getDimentionsByView(activeView);

  const classSet = cx(
    styles.base,
    styles[activeView],
    styles[isOpen ? currentTheme : currentClosedTheme],
    {
      [styles.active]: isOpen,
    },
  );

  return (
    <Hyperlink
      className={classSet}
      dataTestRef={dataTestRef ?? 'NAV_LOGO'}
      tabIndex={!isVisible ? -1 : null}
      title={title}
      url={url}
    >
      <Icon
        className={styles.icon}
        height={height}
        name="aesop"
        tabIndex={-1}
        width={width}
      />
      <ScreenReaderOnly>{label}</ScreenReaderOnly>
    </Hyperlink>
  );
};

export { Logo };
