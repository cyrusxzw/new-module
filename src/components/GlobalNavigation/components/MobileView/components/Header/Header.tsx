import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import {
  useGlobalNavigationContext,
  useGlobalNavigationStateContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';
import { useMobileViewContext } from '../../MobileView.context';
import { useWindowHasResized } from '~/customHooks';
import { Button } from '~/components/Button';
import { Icon } from '~/components/Icon';
import { ScreenReaderOnly } from '~/components/ScreenReaderOnly';
import { Transition } from '~/components/Transition';
import { Logo } from '~/components/GlobalNavigation/components/Logo';
import type { HeaderType } from './Header.types';
import compositionStyles from '../../MobileView.module.css';
import styles from './Header.module.css';

const Header: HeaderType = ({ onClose }) => {
  const {
    isOpen: isMenuOpen,
    setIsOpen: setIsMenuOpen,
    setActiveCollectionId,
    activeCollectionId,
  } = useGlobalNavigationStateContext();

  const { actions, onOpen } = useGlobalNavigationContext();
  const { closedTheme } = useMobileViewContext();
  const currentTheme = useThemeContext(null, 'dark');

  useWindowHasResized();

  const { search, cart, menu, stores } = actions;
  const currentClosedTheme = closedTheme || currentTheme;

  const handleOnSearchClick = () => {
    search.onClick();
    setIsMenuOpen(true);
    setActiveCollectionId(search.id);
  };

  const handleOnCartClick = () => cart.onClick();

  const handleOnMenuButtonClick = () => {
    if (isMenuOpen) {
      if (
        activeCollectionId === search?.id ||
        activeCollectionId === stores?.id
      ) {
        setActiveCollectionId('top');
      } else {
        onClose();
      }
    } else {
      setIsMenuOpen(true);
      onOpen?.();
    }
  };

  const classSet = cx(
    styles.base,
    !isMenuOpen ? styles[`${currentClosedTheme}Closed`] : styles[currentTheme],
  );

  return (
    <>
      <nav
        aria-label="header navigation"
        className={classSet}
        role="navigation"
      >
        <ul className={styles.list}>
          <li className={cx(styles.item, styles.itemLogo)}>
            <Logo closedTheme={closedTheme} />
          </li>

          <li className={styles.item}>
            <Button
              className={cx(styles.action, styles.actionSearch)}
              dataTestRef={search.dataTestRef ?? 'NAV_SEARCH'}
              isInline={true}
              onClick={handleOnSearchClick}
              title={search.title}
            >
              <Icon
                className={styles.searchIcon}
                height={16}
                name="search"
                tabIndex={-1}
                width={16}
              />
            </Button>
            <Transition
              isActive={activeCollectionId === search.id}
              shouldMountOnEnter={true}
              shouldUnmountOnExit={true}
              type="slideLeft"
            >
              <div className={cx(compositionStyles.slide, styles.search)}>
                <search.component />
              </div>
            </Transition>
          </li>

          <li className={styles.item}>
            <Button
              className={cx(
                styles.action,
                styles.actionCart,
                compositionStyles.ornamentalWrapper,
              )}
              dataTestRef={cart.dataTestRef ?? 'NAV_CART'}
              isInline={true}
              onClick={handleOnCartClick}
              title={cart.title}
            >
              <span
                className={cx(
                  compositionStyles.ornamentalHover,
                  styles.ornamentalHover,
                )}
              >
                {cart.label}
              </span>
            </Button>
          </li>

          <li className={cx(styles.item, styles.itemMenu)}>
            <Button
              aria={{ haspopup: true, expanded: isMenuOpen }}
              className={cx(styles.action, styles.actionMenu, {
                [styles.openState]: isMenuOpen,
              })}
              dataTestRef={menu.dataTestRef ?? 'NAV_MOBILE_MENU'}
              isInline={true}
              onClick={handleOnMenuButtonClick}
              title={isMenuOpen ? menu.closeTitle : menu.title}
            >
              <ScreenReaderOnly>
                {isMenuOpen ? menu.closeLabel : menu.label}
              </ScreenReaderOnly>
            </Button>
          </li>
        </ul>
      </nav>
    </>
  );
};

export { Header };
