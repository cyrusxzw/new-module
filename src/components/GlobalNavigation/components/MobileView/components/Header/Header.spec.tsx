/* eslint-disable init-declarations */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Header } from './Header';
import { HeaderFixture } from './Header.fixture';

import {
  GlobalNavigationStateContextProvider,
  useGlobalNavigationStateContext,
  useGlobalNavigationContext,
} from '~/components/GlobalNavigation/GlobalNavigation.context';
import {
  MobileViewContextProvider,
  useMobileViewContext,
} from '~/components/GlobalNavigation/components/MobileView/MobileView.context';

import * as logoFile from '../../../Logo/Logo';

jest.mock('~/components/GlobalNavigation/GlobalNavigation.context');
jest.mock(
  '~/components/GlobalNavigation/components/MobileView/MobileView.context',
);

describe('<GlobalNavigation.MobileView.Header />', () => {
  let logoSpy: jest.SpyInstance;

  const mockActions = {
    stores: { component: 'stores', id: 'stores' },
    search: { component: 'search', id: 'search' },
    cart: 'cart',
    menu: 'menu',
  };

  const testGlobalNavigationContext = {
    actions: mockActions,
    onOpen: jest.fn(),
  };

  const testGlobalNavigationStateContext = {
    activeCollectionId: 'top',
    setActiveCollectionId: jest.fn(),
    isOpen: false,
    setIsOpen: jest.fn(),
  };

  const testMobileViewContext = {
    setActiveNestedCollectionIds: jest.fn(),
    onNestedCollectionClick: jest.fn(),
    activeNestedCollectionIds: ['by skin type'],
    closedTheme: 'dark',
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
    });
    (useGlobalNavigationStateContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationStateContext,
    });
    (useMobileViewContext as jest.Mock).mockReturnValue({
      ...testMobileViewContext,
    });
    logoSpy = jest.spyOn(logoFile, 'Logo').mockImplementation(() => null);
  });

  const TestBed = ({ onClose }: { onClose?: () => void }) => {
    return (
      <GlobalNavigationStateContextProvider>
        <MobileViewContextProvider>
          <Header {...HeaderFixture} onClose={onClose} />;
        </MobileViewContextProvider>
      </GlobalNavigationStateContextProvider>
    );
  };

  it('should render base component correctly', () => {
    const { container } = render(<TestBed />);

    expect(container).toMatchSnapshot();
  });

  it('should use default theme', () => {
    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
    });
    (useMobileViewContext as jest.Mock).mockReturnValue({
      closedTheme: null,
    });
    const { container } = render(<TestBed />);

    expect(container).toMatchSnapshot();
  });

  it('should click menu button to close menu', () => {
    const mockOnCloseFn = jest.fn();
    (useGlobalNavigationStateContext as jest.Mock).mockReturnValue({
      activeCollectionId: 'top',
      setActiveCollectionId: jest.fn(),
      isOpen: true,
      setIsOpen: jest.fn(),
    });
    render(<TestBed onClose={mockOnCloseFn} />);

    const closeBtn = screen.getAllByRole('button')[2];
    closeBtn.click();

    expect(mockOnCloseFn).toHaveBeenCalledTimes(1);
  });

  it('should click menu button to close product search menu', () => {
    (useGlobalNavigationStateContext as jest.Mock).mockReturnValue({
      activeCollectionId: 'search',
      setActiveCollectionId: jest.fn(),
      isOpen: true,
      setIsOpen: jest.fn(),
    });
    render(<TestBed />);

    const { setActiveCollectionId } = useGlobalNavigationStateContext();
    const closeBtn = screen.getAllByRole('button')[2];
    closeBtn.click();

    expect(setActiveCollectionId).toHaveBeenCalledTimes(1);
  });

  it('should click menu button to close store search menu', () => {
    (useGlobalNavigationStateContext as jest.Mock).mockReturnValue({
      activeCollectionId: 'stores',
      setActiveCollectionId: jest.fn(),
      isOpen: true,
      setIsOpen: jest.fn(),
    });
    render(<TestBed />);

    const { setActiveCollectionId } = useGlobalNavigationStateContext();
    const closeBtn = screen.getAllByRole('button')[2];
    closeBtn.click();

    expect(setActiveCollectionId).toHaveBeenCalledTimes(1);
  });

  it('should click menu button to open menu', () => {
    const { onOpen } = useGlobalNavigationContext();
    const { setIsOpen } = useGlobalNavigationStateContext();

    render(<TestBed />);

    const openBtn = screen.getAllByRole('button')[2];
    openBtn.click();

    expect(onOpen).toHaveBeenCalledTimes(1);
    expect(setIsOpen).toHaveBeenCalledTimes(1);
  });

  it('should click menu button to open menu while onOpen is not defined', () => {
    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
      onOpen: null,
    });
    const { onOpen } = useGlobalNavigationContext();
    const { setIsOpen } = useGlobalNavigationStateContext();

    render(<TestBed />);

    const closeBtn = screen.getAllByRole('button')[2];
    closeBtn.click();

    expect(onOpen).toEqual(null);
    expect(setIsOpen).toHaveBeenCalledTimes(1);
  });

  it('should use default data test refs', () => {
    const actions = {
      search: { dataTestRef: 'Custom search dataTestRef', url: '/' },
      cart: { dataTestRef: 'Custom cart dataTestRef', url: '/' },
      menu: { dataTestRef: 'Custom menu dataTestRef', url: '/' },
    };

    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
      actions: actions,
    });

    render(<TestBed />);

    const buttons = screen.getAllByRole('button');
    const navSearch = buttons[0];
    const navCart = buttons[1];
    const navMobileMenu = buttons[2];

    expect(navSearch).toHaveAttribute(
      'data-test-ref',
      'Custom search dataTestRef',
    );
    expect(navCart).toHaveAttribute('data-test-ref', 'Custom cart dataTestRef');
    expect(navMobileMenu).toHaveAttribute(
      'data-test-ref',
      'Custom menu dataTestRef',
    );
  });

  it('should open product search on click', () => {
    const actions = {
      ...mockActions,
      search: {
        dataTestRef: 'Custom search dataTestRef',
        url: '/',
        onClick: jest.fn(),
      },
    };
    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
      actions: actions,
    });

    render(<TestBed />);

    const navSearch = screen.getAllByRole('button')[0];

    navSearch.click();

    expect(actions.search.onClick).toHaveBeenCalledTimes(1);
  });

  it('should open product cart on click', () => {
    const actions = {
      ...mockActions,
      cart: {
        dataTestRef: 'Custom cart dataTestRef',
        url: '/',
        onClick: jest.fn(),
      },
    };

    (useGlobalNavigationContext as jest.Mock).mockReturnValue({
      ...testGlobalNavigationContext,
      actions: actions,
    });

    render(<TestBed />);

    const navCart = screen.getAllByRole('button')[1];

    navCart.click();

    expect(actions.cart.onClick).toHaveBeenCalledTimes(1);
  });
});
