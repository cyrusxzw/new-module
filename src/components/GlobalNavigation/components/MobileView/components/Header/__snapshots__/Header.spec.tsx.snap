// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNavigation.MobileView.Header /> should render base component correctly 1`] = `
<div>
  <nav
    aria-label="header navigation"
    class="base darkClosed"
    role="navigation"
  >
    <ul
      class="list"
    >
      <li
        class="item itemLogo"
      />
      <li
        class="item"
      >
        <button
          class="base dark action actionSearch"
          data-test-ref="NAV_SEARCH"
          type="button"
        >
          <svg
            aria-labelledby="search-1"
            class="base search dark searchIcon"
            data-testid="data-testid-Icon"
            focusable="false"
            height="16"
            role="img"
            style="height: 16px; width: 16px;"
            tabindex="-1"
            viewBox="0 0 16 16"
            width="16"
          >
            <title
              id="search-1"
            >
              search
            </title>
            <g>
              <path
                d="M11.9544 10.8246C12.8838 9.65831 13.3941 8.18223 13.3941 6.68793C13.3941 3.00683 10.3872 0 6.68793 0C2.98861 0 0 3.00683 0 6.68793C0 10.369 3.00683 13.3759 6.68793 13.3759C8.16401 13.3759 9.64009 12.8656 10.8246 11.9362L14.8519 15.9636L16 14.8155L11.9544 10.8246ZM6.68793 11.7904C3.89977 11.7904 1.62187 9.51253 1.62187 6.72437C1.62187 3.93622 3.89977 1.65831 6.68793 1.65831C9.47608 1.65831 11.754 3.93622 11.754 6.72437C11.754 9.51253 9.47608 11.7904 6.68793 11.7904Z"
              />
            </g>
          </svg>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="base dark action actionCart ornamentalWrapper"
          data-test-ref="NAV_CART"
          type="button"
        >
          <span
            class="ornamentalHover ornamentalHover"
          />
        </button>
      </li>
      <li
        class="item itemMenu"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="base dark action actionMenu"
          data-test-ref="NAV_MOBILE_MENU"
          type="button"
        >
          <span
            class="base"
          />
        </button>
      </li>
    </ul>
  </nav>
  ;
</div>
`;

exports[`<GlobalNavigation.MobileView.Header /> should use default theme 1`] = `
<div>
  <nav
    aria-label="header navigation"
    class="base darkClosed"
    role="navigation"
  >
    <ul
      class="list"
    >
      <li
        class="item itemLogo"
      />
      <li
        class="item"
      >
        <button
          class="base dark action actionSearch"
          data-test-ref="NAV_SEARCH"
          type="button"
        >
          <svg
            aria-labelledby="search-2"
            class="base search dark searchIcon"
            data-testid="data-testid-Icon"
            focusable="false"
            height="16"
            role="img"
            style="height: 16px; width: 16px;"
            tabindex="-1"
            viewBox="0 0 16 16"
            width="16"
          >
            <title
              id="search-2"
            >
              search
            </title>
            <g>
              <path
                d="M11.9544 10.8246C12.8838 9.65831 13.3941 8.18223 13.3941 6.68793C13.3941 3.00683 10.3872 0 6.68793 0C2.98861 0 0 3.00683 0 6.68793C0 10.369 3.00683 13.3759 6.68793 13.3759C8.16401 13.3759 9.64009 12.8656 10.8246 11.9362L14.8519 15.9636L16 14.8155L11.9544 10.8246ZM6.68793 11.7904C3.89977 11.7904 1.62187 9.51253 1.62187 6.72437C1.62187 3.93622 3.89977 1.65831 6.68793 1.65831C9.47608 1.65831 11.754 3.93622 11.754 6.72437C11.754 9.51253 9.47608 11.7904 6.68793 11.7904Z"
              />
            </g>
          </svg>
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="base dark action actionCart ornamentalWrapper"
          data-test-ref="NAV_CART"
          type="button"
        >
          <span
            class="ornamentalHover ornamentalHover"
          />
        </button>
      </li>
      <li
        class="item itemMenu"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="base dark action actionMenu"
          data-test-ref="NAV_MOBILE_MENU"
          type="button"
        >
          <span
            class="base"
          />
        </button>
      </li>
    </ul>
  </nav>
  ;
</div>
`;
